spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -35,10 +40,12 @@
  * resolves the channel from the bean factory and, if not present, creates a new channel
  * and adds it to the factory after binding it to the binder. The binder is optionally
  * determined with a prefix preceding a colon.
+ *
  * @author Mark Fisher
  * @author Gary Russell
+ * @author Ilayaperumal Gopinathan
  */
-public class BinderAwareChannelResolver extends BeanFactoryMessageChannelDestinationResolver {
+public class BinderAwareChannelResolver extends BeanFactoryMessageChannelDestinationResolver implements SmartLifecycle {
```

Comment:
I see. I didn't count on the HealthIndicators being produced by an instance method of the same `ChannelBindingAutoConfiguration`. I suppose having a separate bean for holding the dynamic bindings (a collection of them) would break the circular dependency if we injected it into the `BinderAwareChannelResolver` instead?